src/tables.jl

```diff
@@ -38,10 +38,11 @@ function Tables.columns(A::Union{KeyedArray, NdaKa})
     L = hasnames(A) ? (dimnames(A)..., :value) :
         (ntuple(d -> Symbol(:dim_,d), ndims(A))..., :value)
     R = keys_or_axes(A)
-    G = ntuple(ndims(A)) do d
-        vec([rs[d] for rs in Iterators.product(R...)])
-        # _vec(rs[d] for rs in Iterators.product(R...))
-    end
+    R_inds = map(eachindex, R)
+    G = map(
+        (r, d) -> vec([r[rs[d]] for rs in Iterators.product(R_inds...)]),
+        R, ntuple(identity, length(R)),
+    )
```

Comment:
How do you feel about
```suggestion
    G = [
        vec([r[rs[d]] for rs in Iterators.product(R_inds...)])
        for (d, r) in enumerate(R)
    ]
```

Reply:
Not great because it produces a `Vector` rather than a `Tuple`, meaning that type-stability is lost.